component/org.wso2.carbon.identity.casque.authenticator/src/test/java/org.wso2.carbon.identity.casque.authenticator/authenticator/CasqueAuthenticatorTest.java

```diff
@@ -0,0 +1,310 @@
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    private CasqueAuthenticator casqueAuthenticator;
+
+    @Mock
+    private HttpServletRequest httpServletRequest;
+
+    @Mock
+    private HttpServletResponse httpServletResponse;
+
+    @Mock
+    private UserStoreManager userStoreManager;
+
+    @Mock
+    CasqueAuthenticatorServiceDataHolder casqueAuthenticatorServiceDataHolder;
+
+    @Mock
+    RealmService realmService;
+
+    @Mock
+    UserRealm tenantUserRealm;
+
+    @Mock
+    private AuthenticationContext context;
+
+    @Mock
+    private RadiusResponse radiusResponse;
+
+    @Mock
+    private AuthenticatedUser authenticatedUser;
+
+    @Mock
+    private User user;
+
+    @Mock
+    private Map mockMap;
+
+    @BeforeMethod
+    public void setUp() {
+
+        casqueAuthenticator = new CasqueAuthenticator();
+        initMocks(this);
+    }
+
+    @Test(description = "Test case for GetName() method.")
+    public void testGetName() {
+
+        Assert.assertEquals(casqueAuthenticator.getName(),
+                CasqueAuthenticatorConstants.AUTHENTICATOR_NAME);
+    }
+
+    @Test(description = "Test case for RetryAuthenticationEnabled() method.")
+    protected void testRetryAuthenticationEnabled() {
+
+        Assert.assertFalse(casqueAuthenticator.retryAuthenticationEnabled());
+    }
+
+    @Test(description = "Test case for GetFriendlyName() method.")
+    public void testGetFriendlyName() {
+
+        Assert.assertEquals(casqueAuthenticator.getFriendlyName(),
+                CasqueAuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME);
+    }
+
+    @Test(description = "Test case for canHandle() method.")
+    public void testCanHandle() {
+
+        Assert.assertTrue(casqueAuthenticator.canHandle(httpServletRequest));
+
+    }
+
+    @Test(description = "Test case for testGetCasqueTokenId() method.")
+    public void testGetCasqueTokenId() throws Exception {
+
+        mockStatic(CasqueAuthenticatorServiceDataHolder.class);
+        mockStatic(IdentityTenantUtil.class);
+
+        when(CasqueAuthenticatorServiceDataHolder.getInstance()).thenReturn(casqueAuthenticatorServiceDataHolder);
+        when(casqueAuthenticatorServiceDataHolder.getRealmService()).thenReturn(realmService);
+        when(realmService.getTenantUserRealm(IdentityTenantUtil.getTenantIdOfUser(anyString())))
+                .thenReturn(tenantUserRealm);
+        when(tenantUserRealm.getUserStoreManager()).thenReturn(userStoreManager);
+        when(userStoreManager.getUserClaimValues(anyString(), any(String[].class), anyString())).thenReturn(mockMap);
+        when(mockMap.get(anyString())).thenReturn("FFF 000001");
+        Whitebox.setInternalState(CasqueAuthenticatorServiceDataHolder.class, "instance",
+                casqueAuthenticatorServiceDataHolder);
+        Assert.assertEquals(Whitebox.invokeMethod(casqueAuthenticator, "getCasqueTokenId"
+                , ""), "FFF 000001");
```

Comment:
Shall we move this to the above line, if length does not exceed 120?

Reply:
It exceeds 120